logpuzzle.py

```diff
@@ -40,14 +75,25 @@ def download_images(img_urls, dest_dir):
     with an img tag to show each local image file.
     Creates the directory if necessary.
     """
-    # +++your code here+++
-    pass
+
+    os.mkdir(dest_dir)
+
+    with open('index.html', 'w') as index:
+        img_temp_list = []
+        for i, url in enumerate(img_urls):
+            urllib.urlretrieve(url, 'img{}'.format(i))
+            shutil.move('img{}'.format(i), '{}'.format(dest_dir))
+            img_temp_list.append(create_img_template('img{}'.format(i)))
+
+        shutil.move('index.html', dest_dir)
+        index.write(html_template.format('\n'.join(img_temp_list)))
```

Comment:
Joining on newline is preventing the image slices from rendering as a single image. Removing the newline join removes the margins between slices.